code/modules/atmospherics/machinery/components/fusion/hypertorus.dm

```diff
@@ -0,0 +1,1376 @@
+	if(!fusion_started)
+		if(default_deconstruction_screwdriver(user, icon_state_open, icon_state_off, I))
+			return
+	if(default_change_direction_wrench(user, I))
+		return
+	if(default_deconstruction_crowbar(I))
+		return
+	return ..()
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/default_change_direction_wrench(mob/user, obj/item/I)
+	. = ..()
+	if(!.)
+		return
+	if(!anchored)
+		return FALSE
+	var/obj/machinery/atmospherics/node1 = nodes[1]
+	var/obj/machinery/atmospherics/node2 = nodes[2]
+	if(node1)
+		node1.disconnect(src)
+		nodes[1] = null
+		nullifyPipenet(parents[1])
+	if(node2)
+		node2.disconnect(src)
+		nodes[2] = null
+		nullifyPipenet(parents[1])
+
+	SetInitDirections()
+	atmosinit()
+	node1 = nodes[1]
+	if(node1)
+		node1.atmosinit()
+		node1.addMember(src)
+	node2 = nodes[2]
+	if(node2)
+		node2.atmosinit()
+		node2.addMember(src)
+	SSair.add_to_rebuild_queue(src)
+	return TRUE
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core/proc/check_part_connectivity()
+	. = TRUE
+	if(!anchored || panel_open)
+		return FALSE
+
+	for(var/obj/machinery/hypertorus/object in orange(1,src))
+		if(. == FALSE)
+			break
+
+		if(object.panel_open)
+			. = FALSE
+
+		if(istype(object,/obj/machinery/hypertorus/corner))
+			var/dir = get_dir(src,object)
+			if(dir in GLOB.cardinals)
+				. =  FALSE
+			switch(dir)
+				if(SOUTHEAST)
+					if(object.dir != SOUTH)
+						. = FALSE
+				if(SOUTHWEST)
+					if(object.dir != WEST)
+						. =  FALSE
+				if(NORTHEAST)
+					if(object.dir != EAST)
+						. =  FALSE
+				if(NORTHWEST)
+					if(object.dir != NORTH)
+						. =  FALSE
+			corners |= object
+			continue
+
+		if(get_step(object,turn(object.dir,180)) != loc)
+			. =  FALSE
+
+		if(istype(object,/obj/machinery/hypertorus/interface))
+			if(linked_interface && linked_interface != object)
+				linked_interface = null
+				. =  FALSE
+			linked_interface = object
+
+	for(var/obj/machinery/atmospherics/components/unary/hypertorus/object in orange(1,src))
+		if(. == FALSE)
+			break
+
+		if(object.panel_open)
+			. = FALSE
+
+		if(get_step(object,turn(object.dir,180)) != loc)
+			. =  FALSE
+
+		if(istype(object,/obj/machinery/atmospherics/components/unary/hypertorus/fuel_input))
+			if(linked_input && linked_input != object)
+				linked_input = null
+				. =  FALSE
+			linked_input = object
+
+		if(istype(object,/obj/machinery/atmospherics/components/unary/hypertorus/waste_output))
+			if(linked_output && linked_output != object)
+				linked_output = null
+				. =  FALSE
+			linked_output = object
+
+		if(istype(object,/obj/machinery/atmospherics/components/unary/hypertorus/moderator_input))
+			if(linked_moderator && linked_moderator != object)
+				linked_moderator = null
+				. =  FALSE
+			linked_moderator = object
+
+	if(!linked_interface || !linked_input || !linked_moderator || !linked_output || corners.len != 4)
+		. = FALSE
+
+
```

Comment:
Oh jesus it goes on, what's the intent of this proc?

Reply:
as explained above. Those last 2 lines are just to ensure that the proc has done its job